services/galley/src/Galley/Intra/Push.hs

```diff
@@ -107,11 +110,34 @@ data Push
         _pushAsync :: Bool,
         pushOrigin :: UserId,
         pushRecipients :: List1 Recipient,
-        pushJson :: Object
+        pushJson :: Object,
+        pushRecipientListTooLargeToFanout :: Bool
       }
 
 makeLenses ''Push
 
+newPush1Limited :: Data.ResultSetType -> Range 1 Teams.HardTruncationLimit Int32 -> UserId -> PushEvent -> List1 Recipient -> Push
+newPush1Limited recipientsType hardLimit from e rr =
+  Push
+    { _pushConn = Nothing,
+      _pushTransient = False,
+      _pushRoute = Gundeck.RouteAny,
+      _pushNativePriority = Nothing,
+      _pushAsync = False,
+      pushRecipientListTooLargeToFanout = isTooLargeToFanout,
+      pushJson = pushEventJson e,
+      pushOrigin = from,
+      pushRecipients = rr
+    }
+  where
+    -- The second check is redudant but we are paranoid... and it does not hurt
```

Comment:
```suggestion
    -- The second check is redundant but we are paranoid... and it does not hurt
```

Also, the second check could for example kick in when sending a message if the hard truncation limit is configured lower than the maximum conversation size, right? (similarly for other events that gets sent to many people, but not the list of team members).

Reply:
> when sending a message if the hard truncation limit is configured lower than the maximum conversation size

Yes, wanted to make sure we don't blow up, I will triple check that the nr makes sense :)